### PR TITLE
Add retry function, add few tests to cover main functionality

### PR DESCRIPTION
**Pull request by piontkivskyi**

### CODE REVIEW

File: src/test/scala/utils/RetrierTest.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package utils
+
+import scala.concurrent.duration._
+import org.scalatest.FlatSpec
+
+class RetrierTest extends FlatSpec {
+
+  "Retrier " should "run given function if accepted result is not acceptable" in {
```
Comment (azhur) [inline src/test/scala/utils/RetrierTest.scala:8]: run given function if the result is not acceptable :)

File: src/test/scala/utils/RetrierTest.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        1 + n
+      },
+      acceptResult = res => res % 2 == 0,
+      retries = List(0.second, 1.second, 5.second)
+    )
+
+    assert(2 == result)
+  }
+
+  it should "return any value if tries limit is exceeded" in {
```
Comment (azhur) [inline src/test/scala/utils/RetrierTest.scala:23]: more precisely - it should return the last value if retries limit is exceeded

**Review by azhur — COMMENTED** [review]
Great Job!

File: src/main/scala/utils/Retrier.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  final def retry[A](
+                      block: () => A,
+                      acceptResult: A => Boolean,
+                      retries: List[FiniteDuration]
+                    ): A = {
+      val result: A = block()
+      if (acceptResult(result) || retries.isEmpty) {
+        result
+      } else {
+        Thread.sleep(retries.head.toMillis)
```
Comment (nkharabaruk) [inline src/main/scala/utils/Retrier.scala]: it will fail if retries is empty

**Pull request merged**